apps/etterlatte-tidshendelser/src/main/kotlin/no/nav/etterlatte/tidshendelser/AldersovergangerService.kt

```diff
@@ -0,0 +1,32 @@
+package no.nav.etterlatte.tidshendelser
+
+import no.nav.etterlatte.tidshendelser.klient.GrunnlagKlient
+import org.slf4j.LoggerFactory
+
+class AldersovergangerService(
+    private val hendelseDao: HendelseDao,
+    private val grunnlagKlient: GrunnlagKlient,
+) {
+    private val logger = LoggerFactory.getLogger(AldersovergangerService::class.java)
+
+    fun execute(jobb: HendelserJobb) {
+        logger.info("Handling jobb ${jobb.id} med type ${jobb.type} (${jobb.type.beskrivelse})")
+        hendelseDao.oppdaterJobbstatusStartet(jobb)
+
+        if (jobb.type == JobbType.AO_BP20) {
+            val foedselsmaaned = jobb.behandlingsmaaned.minusYears(20)
+
+            val saker = grunnlagKlient.hentSaker(foedselsmaaned = foedselsmaaned)
+            logger.info("Hentet ${saker.size} saker for brukere født i $foedselsmaaned")
+
+            if (saker.isNotEmpty()) {
+                hendelseDao.opprettHendelserForSaker(jobb.id, saker, Steg.IDENTIFISERT_SAK)
```

Comment:
Skal det ikke kjøres oppdaterJobbstatusFerdig når alle hendelser er ferdig?

Reply:
Lytting på hendelser er ikke påbegynt ennå

Reply:
Ah var det du nevnte på standup. Jeg trodde det var for hver enkelte hendelse.

Når en jobb da inneholder mange hendelser så kan det jo gå lang tid før samtlige hendelser er plukket opp eller håndtert?
Hva er egentlig hensikten/gevinsten med å holde en overordna jobb for alle hendelsene åpen?

Reply:
Tenker vi kan fortsette diskusjon der etter hvert, er Helge sin idé :p